froide/foirequest/api/permissions.py

```diff
@@ -0,0 +1,16 @@
+from rest_framework import permissions
+from rest_framework.views import Request
+
+from ..auth import can_write_foirequest
+
+
+class OwnsFoiRequestPermission(permissions.BasePermission):
+    def __init__(self, request_field: str = "request") -> None:
+        self.request_field = request_field
+        super().__init__()
+
+    def has_object_permission(self, request: Request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+
+        return can_write_foirequest(getattr(obj, self.request_field), request)
```

Comment:
Seems to be the case. Somehow worked in my initial testing, but I should probably split this class up for Message and Attachment anyhow, since the check for the `create` method also differs quite a bit.

Reply:
Updated `WriteFoiRequestPermission` like so:

https://github.com/okfde/froide/blob/b5e31e9dce92f6aca4d99200d0acc87d30e90d75/froide/foirequest/api/permissions.py#L11-L19